tools/stats/monitor.py

```diff
@@ -76,78 +59,44 @@ def get_per_process_gpu_info(handle: Any) -> List[Dict[str, Any]]:
-    memoryUse = c_uint64()
-    memoryTot = c_uint64()
-
-    ret = rocmsmi.rsmi_dev_memory_usage_get(device, 0, byref(memoryUse))
-    if rocm_ret_ok(ret):
-        ret = rocmsmi.rsmi_dev_memory_total_get(device, 0, byref(memoryTot))
-        if rocm_ret_ok(ret):
-            return float(memoryUse.value) / float(memoryTot.value)
-    return 0.0
-
-
-def rocm_get_gpu_use(device: int) -> float:
-    percent = c_uint32()
-    ret = rocmsmi.rsmi_dev_busy_percent_get(device, byref(percent))
-    if rocm_ret_ok(ret):
-        return float(percent.value)
-    return 0.0
-
-
-def rocm_get_pid_list() -> List[Any]:
-    num_items = c_uint32()
-    ret = rocmsmi.rsmi_compute_process_info_get(None, byref(num_items))
-    if rocm_ret_ok(ret):
-        buff_sz = num_items.value + 10
-        procs = (rsmi_process_info_t * buff_sz)()
-        procList = []
-        ret = rocmsmi.rsmi_compute_process_info_get(byref(procs), byref(num_items))
-        for i in range(num_items.value):
-            procList.append(procs[i].process_id)
-        return procList
-    return []
-
-
-def rocm_get_per_process_gpu_info() -> List[Dict[str, Any]]:
+def rocm_get_per_process_gpu_info(handle: Any) -> List[Dict[str, Any]]:
+    processes = pyamdsmi.amdsmi_get_gpu_process_list(handle)
     per_process_info = []
-    for pid in rocm_get_pid_list():
-        proc = rsmi_process_info_t()
-        ret = rocmsmi.rsmi_compute_process_info_by_pid_get(int(pid), byref(proc))
-        if rocm_ret_ok(ret):
-            info = {"pid": pid, "gpu_memory": proc.vram_usage}
-            per_process_info.append(info)
+    for p in processes:
+        proc_info = pyamdsmi.amdsmi_get_gpu_process_info(handle, p)
+        info = {
+            "pid": proc_info["pid"],
+            "gpu_memory": proc_info["memory_usage"]["vram_mem"],
+        }
+        per_process_info.append(info)
     return per_process_info
 
 
 if __name__ == "__main__":
     handle = None
     try:
-        pynvml.nvmlInit()
-        handle = pynvml.nvmlDeviceGetHandleByIndex(0)
-    except pynvml.NVMLError:
+        import pynvml  # type: ignore[import]
+
+        try:
+            pynvml.nvmlInit()
+            handle = pynvml.nvmlDeviceGetHandleByIndex(0)
+        except pynvml.NVMLError:
+            pass
+    except ModuleNotFoundError:
         # no pynvml avaliable, probably because not cuda
         pass
-
-    rsmi_handles = []
     try:
-        ret = rocmsmi.rsmi_init(0)
-        rsmi_handles = rocm_list_devices()
-    except Exception:
-        # no rocmsmi available, probably because not rocm
+        import amdsmi as pyamdsmi  # type: ignore[import]
+
+        try:
+            pyamdsmi.amdsmi_init()
+            amdsmi_handle = pyamdsmi.amdsmi_get_processor_handles()[0]
+            amdsmi_tot_vram = pyamdsmi.amdsmi_get_gpu_memory_total(
+                amdsmi_handle, pyamdsmi.AmdSmiMemoryType.VRAM
+            )
+        except pyamdsmi.AmdSmiException:
+            pass
+    except ModuleNotFoundError:
```

Comment:
Same as above.  I think it's better to avoid the nested try/except here.  The result is the same since both except clauses just pass.

```suggestion
        import amdsmi as pyamdsmi  # type: ignore[import]
        pyamdsmi.amdsmi_init()
        amdsmi_handle = pyamdsmi.amdsmi_get_processor_handles()[0]
        amdsmi_tot_vram = pyamdsmi.amdsmi_get_gpu_memory_total(
            amdsmi_handle, pyamdsmi.AmdSmiMemoryType.VRAM
        )
    except (ModuleNotFoundError, pyamdsmi.AmdSmiException):
```